Libs/Hopac/Stream.fs

```diff
@@ -282,7 +282,22 @@ module Stream =
   let joinWith join xs = joinWith' join xs |> memo
 
   let ambAll (xxs: Stream<#Stream<_>>) = joinWith amb' xxs
-  let mergeAll (xxs: Stream<#Stream<_>>) = joinWith merge' xxs
+
+  let rec mergeAll' combined producer =
+    combined ^=> function
+      | Nil -> producer >>= function
+        | Nil -> nilj
+        | Cons (x, xs) -> mergeAll' x xs :> _
+      | Cons (v, vs) -> consj v (mergeAll' vs producer)
+    <|> producer ^=> function
+      | Nil -> combined >>= function
+        | Nil -> nilj
+        | Cons (v, vs) -> consj v (mergeAll' vs nil)
```

Comment:
In the case that the `producer` ends, you should just be able to return `combined`.